### PR TITLE
allow to create Skia drawing context from existing canvas

### PR DESCRIPTION
**Pull request by donandren**
## Checklist

- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by MarchingCube** [comment]
I'm wondering - what is the use case here?

**Comment by akunchev** [comment]
https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skdocument.beginpage?view=skiasharp-1.68.2#SkiaSharp_SKDocument_BeginPage_System_Single_System_Single_

canvas can be created by external api and we can draw to it with avalonia api this way
one example is `SKDocument`
```C#
SKCanvas canvas = SKDocument.BeginPage(...);
//use SKCanvas to render to document
```

**Comment by Gillibald** [comment]
The real question is why we need a SKSurface in the `DrawingContextImpl` wouldn't a SKCanvas be enough?

**Comment by kekekeks** [comment]
@akunchev To make this change we also need some kind of a public API to wrap an existing SKCanvas into a DrawingContext, right now this change might get reverted in further refactorings

@Gillibald it's required by NComposition for backdrop effects like blurbehind

File: src/Skia/Avalonia.Skia/PlatformRenderInterface.cs [hunk 1]
```diff
@@ -145,6 +145,8 @@ public IRenderTarget CreateRenderTarget(IEnumerable<object> surfaces)
             {
                 if (surface is IFramebufferPlatformSurface framebufferSurface)
                     return new FramebufferRenderTarget(framebufferSurface);
+                if (surface is IExternalCanvasSurface canvas)
```
Comment (donandren) [inline src/Skia/Avalonia.Skia/PlatformRenderInterface.cs:148]: @kekekeks some api like this?
`IPlatformRenderInterface.CreateRenderTarget` with interface for Canvas and Dpi

Comment (kekekeks) [reply src/Skia/Avalonia.Skia/PlatformRenderInterface.cs:148]: Anything in AvaloniaLocator is technically a private API. So I'd rather have some class with a `DrawingContext WrapSkiaCanvas(Canvas canvas, Vector dpi)` static method in `Avalonia.Skia` assembly.

File: src/Skia/Avalonia.Skia/Helpers/DrawingContextHelper.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        /// <summary>
+        /// Wrap Skia canvas in drawing context so we can use Avalonia api to render to external skia canvas
+        /// this is useful in scenarios where canvas is not controlled by application, but received from another non avalonia api
+        /// like: SKCanvas canvas = SKDocument.BeginPage(...);
+        /// </summary>
+        /// <param name="canvas"></param>
+        /// <param name="dpi"></param>
+        /// <param name="visualBrushRenderer"></param>
+        /// <returns>DrawingContext</returns>
+        public static IDrawingContextImpl WrapSkiaCanvas(SKCanvas canvas, Vector dpi, IVisualBrushRenderer visualBrushRenderer = null)
```
Comment (kekekeks) [inline src/Skia/Avalonia.Skia/Helpers/DrawingContextHelper.cs:18]: Should we return the `Impl` here? It's also technically a private API.

Comment (donandren) [reply src/Skia/Avalonia.Skia/Helpers/DrawingContextHelper.cs:18]: I'll be more happy to return the impl, here, as it's a public interface and suits for my use case. 
But if you strongly believe we shouldn't return the impl, i can change it to return `DrawingContext` and will use some workarounds to get the impl, as in my scenario i need and use the impl.

Comment (Gillibald) [reply src/Skia/Avalonia.Skia/Helpers/DrawingContextHelper.cs:18]: This is some kind of "I know what I am doing API" and `DrawingContext` can be constructed from a `IDrawingContextImpl` so this should be fine.

**Review by kekekeks — APPROVED** [review]

**Pull request merged**